src/app/components/RadioButton/index.tsx

```diff
@@ -0,0 +1,103 @@
+import React, { useCallback, useMemo, useState } from 'react';
+import classNames from 'classnames';
+import styles from './index.module.scss';
+import {
+  MinimumTwoEntries,
+  RadioButtonColor,
+  RadioButtonSize,
+  RadioButtonStyle,
+} from './types';
+
+type RadioButtonProps = {
+  merged: boolean;
+  entries: MinimumTwoEntries;
+  onChange: (value: any) => void;
+  size?: RadioButtonSize;
+  color?: RadioButtonColor;
+  style?: RadioButtonStyle;
+  disabled?: boolean;
+  selected?: any;
+  fullWidth?: boolean;
+};
+
+export const RadioButton: React.FC<RadioButtonProps> = ({
+  merged,
+  entries,
+  onChange,
+  size = RadioButtonSize.md,
+  color = RadioButtonColor.secondary,
+  style = RadioButtonStyle.normal,
+  disabled,
+  selected = undefined,
+  fullWidth,
+}) => {
+  const [isSelected, setIsSelected] = useState(selected);
+
+  const handleOnChange = useCallback(
+    value => {
+      setIsSelected(value);
+      onChange(value);
+    },
+    [onChange],
+  );
+
+  const classNameComplete = useMemo(
+    () =>
+      classNames(
+        styles.button,
+        size && styles[size],
+        color && styles[color],
+        style && styles[style],
```

Comment:
To me it would be better to have styles mapped for each enum value enforcing typecheck, so it would show error when new size/color/style is added to enums but classes are not added:

```typescript
const colorClassMap: Record<RadioButtonColor, string> = {
   // ...  
  [RadioButtonSize.tradeLong]: 'tw-text-trade-long tw-bg-trade-long tw-border-trade-long',
   // ...
};
```

Reply:
yes, does make sense